src/raw/mod.rs

```diff
@@ -208,47 +209,34 @@ fn bucket_mask_to_capacity(bucket_mask: usize) -> usize {
     }
 }
 
+#[cfg_attr(feature = "inline-more", inline)]
+fn align_size(size: usize, align: usize) -> Option<usize> {
+    Some(size.checked_add(align - 1)? & !(align - 1))
+}
+
 // Returns a Layout which describes the allocation required for a hash table,
-// and the offset of the buckets in the allocation.
+// and the offset of the control bytes in the allocation.
+// (the offset is also one past last element of data table)
 ///
 /// Returns `None` if an overflow occurs.
 #[cfg_attr(feature = "inline-more", inline)]
-#[cfg(feature = "nightly")]
 fn calculate_layout<T>(buckets: usize) -> Option<(Layout, usize)> {
     debug_assert!(buckets.is_power_of_two());
 
-    // Array of buckets
-    let data = Layout::array::<T>(buckets).ok()?;
+    let align = cmp::max(mem::align_of::<T>(), Group::WIDTH);
 
-    // Array of control bytes. This must be aligned to the group size.
-    //
-    // We add `Group::WIDTH` control bytes at the end of the array which
-    // replicate the bytes at the start of the array and thus avoids the need to
-    // perform bounds-checking while probing.
-    //
-    // There is no possible overflow here since buckets is a power of two and
-    // Group::WIDTH is a small number.
-    let ctrl = unsafe { Layout::from_size_align_unchecked(buckets + Group::WIDTH, Group::WIDTH) };
+    unsafe {
+        let data = Layout::from_size_align_unchecked(
+            align_size(mem::size_of::<T>() * buckets, align)?, align);
 
-    ctrl.extend(data).ok()
-}
+        let ctrl = Layout::from_size_align_unchecked(
+            align_size(buckets + Group::WIDTH, align)?, align);
```

Comment:
You don't actually need to align the size of `ctrl` here since it's at the end of the allocation.